metapool/prep.py

```diff
@@ -0,0 +1,424 @@
+import re
+import os
+import gzip
+import warnings
+import pandas as pd
+
+from glob import glob
+from datetime import datetime
+from string import ascii_letters, digits
+
+# TODO: Ideally these should be imported from Qiita
+PREP_COLUMNS = ['sample_name', 'experiment_design_description',
+                'library_construction_protocol', 'platform', 'run_center',
+                'run_date', 'run_prefix', 'sequencing_meth', 'center_name',
+                'center_project_name', 'instrument_model', 'runid',
+                'sample_plate', 'sample_well', 'i7_index_id', 'index',
+                'i5_index_id', 'index2', 'lane', 'sample_project',
+                'well_description']
+
+# put together by Gail, based on the instruments we know of
+INSTRUMENT_LOOKUP = pd.DataFrame({
+    'A00953': {'machine prefix': 'A', 'Vocab': 'Illumina NovaSeq 6000',
+               'Machine type': 'NovaSeq', 'run_center': 'IGM'},
+    'A00169': {'machine prefix': 'A', 'Vocab': 'Illumina NovaSeq 6000',
+               'Machine type': 'NovaSeq', 'run_center': 'LJI'},
+    'M05314': {'machine prefix': 'M', 'Vocab': 'Illumina MiSeq',
+               'Machine type': 'MiSeq', 'run_center': 'KLM'},
+    'K00180': {'machine prefix': 'K', 'Vocab': 'Illumina HiSeq 4000',
+               'Machine type': 'HiSeq', 'run_center': 'IGM'},
+    'D00611': {'machine prefix': 'D', 'Vocab': 'Illumina HiSeq 2500',
+               'Machine type': 'HiSeq/RR', 'run_center': 'IGM'},
+    'MN01225': {'machine prefix': 'MN', 'Vocab': 'Illumina MiniSeq',
+                'Machine type': 'MiniSeq', 'run_center': 'CMI'}}).T
+
+
+def parse_illumina_run_id(run_id):
+    """Parse a run identifier
+
+    Parameters
+    ----------
+    run_id: str
+        The name of a run
+
+    Returns
+    -------
+    str:
+        When the run happened (YYYY-MM-DD)
+    str:
+        Instrument code
+    """
+
+    # of the form
+    # YYMMDD_machinename_XXXX_FC
+    tokens = run_id.split('_', 1)
+
+    if len(tokens) != 2:
+        raise ValueError('Unrecognized run identifier format "%s". The '
```

Comment:
This error message doesn't make sense compared to the split above, right? Basically, the split takes the first _ so the format can also be: XX_YY or even _XX ...

Reply:
Good call, I've replaced this for a more stringent regex.